### PR TITLE
add Cluster API client

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards Node Pools. Adding the CMA client here.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -136,9 +136,9 @@ required = [
   name = "k8s.io/apiextensions-apiserver"
   version = "kubernetes-1.13.2"
 
-[[constraint]]
+[[override]]
   name = "k8s.io/apimachinery"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.13.1"
```
Comment (xh3b4sd) [inline Gopkg.toml]: I couldn't make it work other than with the override and going back one patch version. That is some weird upstream thing with the currently available version.

File: Gopkg.toml [hunk 2]
```diff
@@ -148,6 +148,10 @@ required = [
   name = "k8s.io/client-go"
   version = "kubernetes-1.13.2"
 
+[[constraint]]
+  name = "sigs.k8s.io/cluster-api"
+  version = "=v0.1.0"
```
Comment (xh3b4sd) [inline Gopkg.toml]: Pinning sharp.

File: service/controller/clusterapi/cluster.go [hunk 3]
```diff
@@ -22,6 +24,7 @@ import (
 )
 
 type ClusterConfig struct {
+	CMAClient    clientset.Interface
```
Comment (xh3b4sd) [inline service/controller/clusterapi/cluster.go]: Upstream it is called "Cluster Management API", short CMA.

File: service/controller/clusterapi/drainer.go [hunk 4]
```diff
@@ -65,7 +68,7 @@ func NewDrainer(config DrainerConfig) (*Drainer, error) {
 	{
 		c := informer.Config{
 			Logger:  config.Logger,
-			Watcher: config.G8sClient.ProviderV1alpha1().AWSConfigs(""),
+			Watcher: config.CMAClient.ClusterV1alpha1().Clusters(corev1.NamespaceAll),
```
Comment (xh3b4sd) [inline service/controller/clusterapi/drainer.go]: Watching clusters here now. This breaks all resources, which I focus on get working next.

File: Gopkg.toml (hunk 1 above)
Comment (tuommaki) [reply Gopkg.toml]: This is something that `cluster-api` dependency brought?

Comment (xh3b4sd) [reply Gopkg.toml]: Yes.

**Pull request merged**